### DISCUSSION TITLE
[FEATURE] XML attribute

### DISCUSSION THREAD

**Discussion by InfinityLoopLabs** [saltstack/salt#67451] [Feature Requests]
I am trying to use this command:
`
salt '*' xml.set_attribute C:\\service.exe.config ".//connectionStrings//add[@name='DefaultConnection']" connectionString "Data Source=SEPVVDB-ER01;Initial Catalog=Pol_EX; Rewrite=true;"
`
You can see the document before and after this command, it changes structurally.
![271269-alexfas01_3840x2400](https://user-images.githubusercontent.com/58587081/89341324-cae45480-d6a9-11ea-9279-13846dd2dc77.jpg)

Are there any options for what I am doing wrong?

**Comment by sagetherage** [comment]
@Morozzko after you run this is the XML no longer valid?

**Comment by Darkformer** [comment]
@sagetherage , let me continue the thread - yes, the xml is no longer valid. For starters, it deletes the first line (the whole xml declaration) and secondly it adds namespaces to elements which prevents the .net framework from parsing this config.

The possible reason would be that instead of actually just changing the attribute, xml module [reads the whole file in the ElementTree](https://github.com/saltstack/salt/blob/1e6bc974f7def4f852f7e9604173e65f8b4e0ad9/salt/modules/xml.py#L96) and [then re-writes the whole tree to the file](https://github.com/saltstack/salt/blob/1e6bc974f7def4f852f7e9604173e65f8b4e0ad9/salt/modules/xml.py#L102).

**Comment by cmcmarrow** [comment]
Like @Darkformer said we rewire the XML file. We should add a flag that only sets the element and nothing else. This will help preserve visibility and avoid confusion in the future.
